crates/notion-core/src/project.rs

```diff
@@ -223,42 +223,58 @@ impl Project {
         path
     }
 
-    /// Writes the specified version of Node or Yarn to the `toolchain` in package.json.
-    pub fn pin(&self, distro_version: &DistroVersion) -> Fallible<()> {
-        match distro_version {
-            DistroVersion::Node(runtime, npm) => {
-                // prevent writing the npm version if it is equal to the default version
-                let default_npm = load_default_npm_version(&runtime).ok();
-                let npm_str = if Some(npm.clone()) == default_npm {
-                    None
-                } else {
-                    Some(npm.to_string())
-                };
-
-                let toolchain = serial::ToolchainSpec::new(
-                    runtime.to_string(),
-                    npm_str,
-                    self.manifest().yarn_str().clone(),
-                );
-                Manifest::update_toolchain(toolchain, self.package_file())?;
-            }
-            DistroVersion::Yarn(version) => {
-                if let Some(platform) = self.manifest().platform() {
-                    let toolchain = serial::ToolchainSpec::new(
-                        platform.node_runtime.to_string(),
-                        platform.npm.as_ref().map(|npm| npm.to_string()),
-                        Some(version.to_string()),
-                    );
-                    Manifest::update_toolchain(toolchain, self.package_file())?;
-                } else {
-                    throw!(ErrorDetails::NoPinnedNodeVersion);
-                }
-            }
-            // ISSUE (#175) When we can `notion install npm` then it can be pinned in the toolchain
-            DistroVersion::Npm(_) => unimplemented!("cannot pin npm in \"toolchain\""),
-            DistroVersion::Package(_, _) => throw!(ErrorDetails::CannotPinPackage),
+    /// Writes the specified version of Node to the `toolchain.node` key in package.json.
+    pub fn pin_node(&self, node_version: &NodeVersion) -> Fallible<()> {
+        // prevent writing the npm version if it is equal to the default version
+        let default_npm = load_default_npm_version(&node_version.runtime).ok();
+        let npm_str = if Some(node_version.npm.clone()) == default_npm {
+            None
+        } else {
+            Some(node_version.npm.to_string())
+        };
+
+        let toolchain = serial::ToolchainSpec::new(
+            node_version.runtime.to_string(),
+            npm_str,
+            self.manifest().yarn_str().clone(),
+        );
+        Manifest::update_toolchain(toolchain, self.package_file())?;
+        println!(
+            "Pinned node version {} in package.json",
+            node_version.runtime
```

Comment:
Outputting the `npm` version as well when `node` is pinned is useful, because that is what will be used for `npm` until the user pins it themselves. Also, since we currently don't support pinning `npm`, showing the user which version will be used is nice feature.

Reply:
Good idea, I'll add that